From 0bbe10313e0b61812082d47431e8648f9df48f15 Mon Sep 17 00:00:00 2001
From: Jeff King <peff@peff.net>
Date: Sat, 4 Mar 2023 05:31:47 -0500
Subject: [PATCH 0999/2871] parse-options: use
 prefix_filename_except_for_dash() helper

Since our fix_filename()'s only remaining special case is handling "-",
we can use the newly-minted helper function that handles this already.

Signed-off-by: Jeff King <peff@peff.net>
Signed-off-by: Junio C Hamano <gitster@pobox.com>
---
 parse-options.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/parse-options.c b/parse-options.c
index 70f827a073..46fe30ddd4 100644
--- a/parse-options.c
+++ b/parse-options.c
@@ -63,10 +63,8 @@ static void fix_filename(const char *prefix, char **file)
 {
 	if (!file || !*file)
 		; /* leave as NULL */
-	else if (!strcmp("-", *file))
-		*file = xstrdup(*file);
 	else
-		*file = prefix_filename(prefix, *file);
+		*file = prefix_filename_except_for_dash(prefix, *file);
 }
 
 static enum parse_opt_result opt_command_mode_error(
-- 
2.39.3 (Apple Git-145)

